### ISSUE TITLE
Requiring modules on Node requires ".js" suffix when the last word is "null".

### ISSUE THREAD

**Issue by chiefGui** [joyent/node#25089]
Hey, guys!

A very specific and weird error happened to me this morning, and I waste a considerable time trying to find out what's happening.

I'm writing tests with Jasmine and I have a fixture called **get_assets_response_null.js**. To call it through my spec, I do `require ('fixtures/get_assets_response_null')` as always. "Suddenly", the following error came out:

> ERROR in ./spec/fixtures/get_assets_response_null
Module build failed: Error: ENOENT, open '/spec/fixtures/get_assets_previous_page_null'
  at Error (native)

I thought "_whoa! wth?_". I tried some fixes but nothing worked. Finally, with my hopes in low level, I decided to place `.js` suffixing the name of the file on the `require` method this way:

```js
require ('./fixtures/get_assets_response_null.js')
```

then boom! It worked!

It was strange, yes, then a coworker gave me a very exotic idea: "_have you tried to remove the `null` word from the file name?_". Then, I removed `null`, leaving the file name as `get_assets_response.js` and changed my `require` accordingly:

```js
require ('./fixtures/get_assets_response')
```

And voi là! It worked like a charm!

In short, is this a bug?

-------
The specifications:

```shell
➜  js-sdk git:(feature/assets) ✗ node -v
v0.12.0
```

**Comment by silverwind** [comment]
`require ('fixtures/get_assets_response_null')` is not a valid way to require a relative module, these *need* to start with `./`. Also I can't get it to reproduce in io.js at least (the module code should still be pretty much the same as in node), could you provide clear steps to reproduce?

**Comment by cjihrig** [comment]
I'm also unable to reproduce the trailing `null` issue.

**Comment by CGavrila** [comment]
I reproduced it earlier today on v0.10.38 if I'm not mistaken. Didn't double check, so don't take my word for it.

I don't have access to the code right now, but I'll come back later with some steps to reproduce.

**Comment by ravi** [comment]
@silverwind since when is your claim true? I use v0.10.x and I use require('a/b/c') all the time with NODE_PATH and it works perfectly fine.

**Comment by silverwind** [comment]
Oh right, I suppose with NODE_PATH it's fine (since that's not relative in that case anymore).

**Comment by chiefGui** [comment]
@silverwind was a typo writing the issue. I used `./`.

**Comment by jasnell** [comment]
Using v0.12 and v0.10.39-pre, with a very quick test case, `touch test_null.js && node -e 'require("./test_null");' && rm test_null.js` demonstrates no issue. The `test_null` module is found with no problem.

**Comment by CGavrila** [comment]
@jasnell: can't re-reproduce it now, I must have done something wrong when I did. As I said, I did it really quickly, so I think I might have missed the `./` before the filename or something like that.

**Comment by cjihrig** [comment]
No one seems to be able to reproduce it, and honestly, it seems more like user error than anything else. If anyone finds a way to consistently reproduce this, please leave a test case here, and I'll gladly reopen.
